bindu/extensions/semantic_memory/embeddings.py

```
@@ -0,0 +1,36 @@
+    # 🔥 TEST-SAFE FALLBACK
+    if not client:
+        return [0.0] * 1536  # matches embedding size
+
+    try:
+        response = client.embeddings.create(
+            model="text-embedding-3-small",
+            input=text,
+        )
+        return response.data[0].embedding
+
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)